test/unit/ngsiv2/HTTP_reveice_measures-test2.js

```diff
@@ -0,0 +1,391 @@
+            // device provisioning functionality. Appropriate verification is done in tests under
+            // provisioning folder of iotagent-node-lib
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
+                .reply(204);
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post(
+                    '/v2/entities/e0130101/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantMeasures.json')
+                )
+                .query({ type: 'sensor' })
+                .reply(204);
+
+            iotaJson.stop(function() {
+                config.iota.timestamp = true;
+                config.compressTimestamp = false;
+                iotaJson.start(config, function() {
+                    request(provisionOptions, function(error, response, body) {
+                        done();
+                    });
+                });
+            });
+        });
+
+        afterEach(function() {
+            config.iota.timestamp = false;
+            config.compressTimestamp = true;
+        });
+
+        it('should send its value to the Context Broker', function(done) {
+            request(optionsMeasure, function(error, result, body) {
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When a POST measure arrives for an unprovisioned device', function() {
+        var optionsMeasure = {
+            url: 'http://localhost:' + config.http.port + '/iot/json',
+            method: 'POST',
+            json: [
+                {
+                    humidity: '33',
+                    temperature: '89'
+                },
+                {
+                    humidity: '32',
+                    temperature: '87'
+                }
+            ],
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            },
+            qs: {
+                i: 'JSON_UNPROVISIONED',
+                k: 'KL223HHV8732SFL1'
+            }
+        };
+        // This mock does not check the payload since the aim of the test is not to verify
+        // device provisioning functionality. Appropriate verification is done in tests under
+        // provisioning folder of iotagent-node-lib
+        beforeEach(function(done) {
+            contextBrokerUnprovMock = nock('http://unexistentHost:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
```

Comment:
Similar case to https://github.com/telefonicaid/iotagent-json/pull/396/files#r344290893